qiskit_nature/runtime/vqe_program.py

```diff
@@ -41,342 +45,68 @@ def __init__(
-            raise ValueError("The provider has not been set.")
-
-        # try to convert the operators to a PauliSumOp, if it isn't already one
-        operator = _convert_to_paulisumop(operator)
-        if aux_operators is not None:
-            aux_operators = [_convert_to_paulisumop(aux_op) for aux_op in aux_operators]
-
-        # combine the settings with the given operator to runtime inputs
-        inputs = {
-            "operator": operator,
-            "aux_operators": aux_operators,
-            "ansatz": self.ansatz,
-            "optimizer": self.optimizer,
-            "initial_point": self.initial_point,
-            "shots": self.shots,
-            "measurement_error_mitigation": self.measurement_error_mitigation,
-            "store_intermediate": self.store_intermediate,
-        }
-
-        # define runtime options
-        options = {"backend_name": self.backend.name()}
-
-        # send job to runtime and return result
-        job = self.provider.runtime.run(
-            program_id=self.program_id,
-            inputs=inputs,
-            options=options,
-            callback=self._wrap_vqe_callback(),
-        )
-        # print job ID if something goes wrong
-        try:
-            result = job.result()
-        except Exception as exc:
-            raise RuntimeError(f"The job {job.job_id()} failed unexpectedly.") from exc
-
-        # re-build result from serialized return value
-        vqe_result = VQEProgramResult()
-        vqe_result.job_id = job.job_id()
-        vqe_result.cost_function_evals = result.get("cost_function_evals", None)
-        vqe_result.eigenstate = result.get("eigenstate", None)
-        vqe_result.eigenvalue = result.get("eigenvalue", None)
-        vqe_result.aux_operator_eigenvalues = result.get("aux_operator_eigenvalues", None)
-        vqe_result.optimal_parameters = result.get("optimal_parameters", None)
-        vqe_result.optimal_point = result.get("optimal_point", None)
-        vqe_result.optimal_value = result.get("optimal_value", None)
-        vqe_result.optimizer_evals = result.get("optimizer_evals", None)
-        vqe_result.optimizer_time = result.get("optimizer_time", None)
-        vqe_result.optimizer_history = result.get("optimizer_history", None)
+        result = super().compute_minimum_eigenvalue(operator, aux_operators)
+
+        # convert to previous result type
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore', category=DeprecationWarning)
+            vqe_result = VQEProgramResult()
+
+        for attr in [
+            "job_id",
+            "cost_function_evals",
+            "eigenstate",
+            "eigenvalue",
+            "aux_operator_eigenvalues",
+            "optimal_parameters",
+            "optimal_point",
+            "optimal_value",
+            "optimizer_evals",
+            "optimizer_time",
+            "optimizer_history",
+        ]:
+            setattr(vqe_result, attr, getattr(result, attr))
```

Comment:
Is this manual approach really necessary? Given that `VQEProgramResult` derives from the new `VQERuntimeResult`, a simple `cast` should be sufficient, right?

Reply:
I'm not sure -- if we use `typing.cast` the actual type is not changed right? So if someone does a typecheck it would fail. But maybe there's a different way of casting that I'm not aware of 😄

Reply:
Correct a `cast` is more like a typehint in saying that this is the type it should be considered as - it does not change what it actually is.

Reply:
Okay let's put it another way then: we never declare the return-type to be a `VQEProgramResult`. We only require it to be a `MinimumEigensolverResult`. Thus, I don't see the need for wrapping a `VQERuntimeResult` with a `VQEProgramResult`. We can simply return the former and deprecate the latter.

Reply:
Even if we didn't document it as precisely we still change the return type, so to be safe I'd prefer to still return the old type 🙂

Reply:
Alright. May I still suggest the use of `AlgorithmResult.combine()` to handle the updating for us?